app/src/main/java/com/example/frogweather/MainActivity.kt

```diff
@@ -1,16 +1,103 @@
+import androidx.core.content.ContextCompat
 import com.example.frogweather.databinding.ActivityMainBinding
+import com.example.frogweather.data.MyLocation
+import com.example.frogweather.data.Sys
+import com.example.frogweather.ui.FrogWeatherApplication
+import com.example.frogweather.ui.SettingsViewModel
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationServices
+import com.google.android.gms.location.Priority
+import com.google.android.gms.tasks.CancellationToken
+import com.google.android.gms.tasks.CancellationTokenSource
+import com.google.android.gms.tasks.OnTokenCanceledListener
+
 
 class MainActivity : AppCompatActivity() {
     private lateinit var binding: ActivityMainBinding
+    private lateinit var fusedLocationClient: FusedLocationProviderClient
+    private val settingsViewModel: SettingsViewModel by viewModels {
+        SettingsViewModel.SettingsViewModelFactory(
+            (application as FrogWeatherApplication)
+        )
+    }
+    private val requestPermissionLauncher =
+        registerForActivityResult(
+            ActivityResultContracts.RequestPermission()
+        ) { isGranted: Boolean ->
+            if (isGranted) {
+                updateLocation()
+            }
+        }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(this)
+
+        when {
+            ContextCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACCESS_COARSE_LOCATION
+            ) == PackageManager.PERMISSION_GRANTED -> {
+                updateLocation()
+            }
+            shouldShowRequestPermissionRationale(Manifest.permission.ACCESS_COARSE_LOCATION) -> {
+                showRequestPermissionDialog()
+            }
+            else -> {
+                requestPermissionLauncher.launch(
+                    Manifest.permission.ACCESS_COARSE_LOCATION
+                )
+            }
+        }
+    }
+
+    private fun showRequestPermissionDialog() {
+        val alertDialog = AlertDialog.Builder(this, R.style.SettingsDialogTheme)
+        alertDialog.setTitle(getString(R.string.lbl_location_settings_dialog_title))
+        alertDialog.setMessage(getString(R.string.msg_location_permisson_dialog))
+        alertDialog.setNegativeButton(getString(R.string.lbl_cancel_button)) { _, _ -> }
+        alertDialog.setPositiveButton(getString(R.string.lbl_settings_button)) { _, _ ->
+            val intent = Intent()
+            intent.action = Settings.ACTION_APPLICATION_DETAILS_SETTINGS
+            val uri = Uri.fromParts("package", packageName, null)
+            intent.data = uri
+            startActivity(intent)
+        }
+        val alert = alertDialog.create()
```

Comment:
No need for variable here, use alertDialog.create().show()

Reply:
done